code/Controllers/CMSMain.php

```diff
@@ -891,9 +1013,11 @@ public function ExtraTreeTools()
      * and can be modified on extensions through updateSearchContext
      *
      * @return \SilverStripe\ORM\Search\SearchContext
+     * @deprecated 5.4.0 Use SilverStripe\CMS\Model\SiteTree::getDefaultSearchContext() instead.
      */
     public function getSearchContext()
     {
+        Deprecation::notice('5.4.0', 'Use ' . SiteTree::class . '::getDefaultSearchContext() instead.');
```

Comment:
getDefaultSearchContext() is not quite equivalent as it does not call $this->extend('updateSearchContext', $context);

Reply:
What would you like the message to be?
"Will be removed without equivalent functionality" isn't correct because `getDefaultSearchContext()` is _effectively_ equivalent, just not 100% identical.
Maybe "Will be replaced with SiteTree::getDefaultSearchContext()`?

Reply:
`Will be replaced with SiteTree::getDefaultSearchContext()` makes sense

Should ensure that the/an extension hook makes it way into the CMS 6 version

Reply:
> You should also ensure that the/an extension hook makes it way into the CMS 6 version

I don't see any value in that - devs can just override `getDefaultSearchContext()` in their subclasses of `SiteTree`.